lib/src/ui/codelessly_widget.dart

```diff
@@ -453,18 +490,30 @@ class _CodelesslyWidgetState extends State<CodelesslyWidget> {
                 );
           }
 
-          if (!model.layouts.containsKey(layoutID)) {
+          if (!model.layouts.containsKey(effectiveLayoutID)) {
             return _effectiveLoadingBuilder?.call(context) ??
                 const CodelesslyLoadingScreen();
           }
 
+          // Set the canvas ID if it is not already set. (This is for the first
+          // time the layout is loaded. For subsequent loads, the canvas ID is
+          // set in the didChangeDependencies method.) e.g. when media query
+          // changes.
+          canvasID ??= getCanvasIDForLayoutGroup(
+              effectiveLayoutID, model, MediaQuery.sizeOf(context));
+
           final layoutWidget = Material(
             clipBehavior: Clip.none,
             type: MaterialType.transparency,
             child: CodelesslyLayoutBuilder(
-              key: ValueKey(layoutID),
+              // This key is important to ensure that the layout is rebuilt
+              // from scratch whenever the layout ID changes, not when the
+              // canvas ID changes.
```

Comment:
Please be more verbose here. Why do we not want to rebuild when the canvas ID changes and only when layout ID changes?